packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -419,35 +320,56 @@ export const Audio = () => {
     }
   }
 
-  return (
-    <View style={styles.backgroundVideo}>
-      {source && (
-        <Video
-          // @ts-ignore: type: m3u8 is actually a valid prop override
-          source={source}
-          ref={videoRef}
-          playInBackground
-          playWhenInactive
-          allowsExternalPlayback={false}
-          audioOnly
-          // Mute playback if we are casting to an external source
-          muted={isCasting}
-          onError={handleError}
-          onEnd={() => {
-            setDuration(0)
-            pause()
-            onNext()
-          }}
-          progressUpdateInterval={100}
-          onLoad={(payload) => {
-            setDuration(payload.duration)
-          }}
-          onProgress={onProgress}
-          repeat={repeatMode === RepeatMode.SINGLE}
-          paused={!playing}
-          // onBuffer={this.onBuffer}
-        />
-      )}
-    </View>
+  const currentUriRef = useRef<string | null>(null)
+  const isPodcastRef = useRef<boolean>(false)
+
+  const handleSourceChange = useCallback(
+    async (newSource: { uri: string; type?: string }) => {
+      const newUri = newSource.uri
+      if (currentUriRef.current !== newUri) {
+        currentUriRef.current = newUri
+        const imageUrl = trackImageSource?.source[2].uri ?? DEFAULT_IMAGE_URL
+        await TrackPlayer.reset()
+        await TrackPlayer.add({
+          url: newUri,
+          type: newSource.type === 'm3u8' ? TrackType.HLS : TrackType.Default,
```

Comment:
we try this with the feature flag off? that's amazing if it does work with the base64 uris!

Reply:
It works with both 🙌

Reply:
awesome!